[Bug] return struct with strings

## Description
Codegen produces invalid C code when returning a struct that contains string fields. The generated C code has a syntax error: "expected ';' before 'return'".

## Steps to Reproduce
1. Create a file with a function that returns a struct containing string fields
2. Run `urusc file.urus`

## Expected Behavior
Should compile successfully and generate valid C code.

## Actual Behavior
Compilation fails with C syntax error.

## Code (if relevant)
```rust
struct Data {
    name: str;
    value: str;
}

fn create_data(a: str, b: str): Data {
    let d: Data = Data { name: a, value: b };
    return d;
}

fn main(): void {
    let d: Data = create_data("hello", "world");
    print(d.name);
}
```

## Error Message
```
_urus_tmp.c:539:5: error: expected ';' before 'return'
  539 |     return _urus_ret_1;
      |     ^~~~~~
Compilation failed.
```

## Environment
- OS: Linux (Arch)
- GCC version: 15.2.1
- URUS version: 0.2.2 (F)-Fixed

--- a/compiler/src/codegen.c
+++ b/compiler/src/codegen.c
@@ -698,7 +698,7 @@ static void gen_stmt(CodeBuf *buf, AstNode *node) {
             if (type_needs_rc(t) && node->as.return_stmt.value->kind == NODE_IDENT) {
                 // the identifier must be set to NULL to prevent destructed from URUS_RAII()
                 emit_indent(buf);
-                emit(buf, "%s = NULL // move to _urus_ret_%d\n", node->as.return_stmt.value->as.ident.name, tmp);
+                emit(buf, "%s = NULL; // move to _urus_ret_%d\n", node->as.return_stmt.value->as.ident.name, tmp);
             }
 
             emit_indent(buf);

